File: app/src/main/java/com/bzh/dytt/di/FragmentBuildersModule.java
package com.bzh.dytt.di;

import com.bzh.dytt.ui.ColorHuntFragment;
import com.bzh.dytt.ui.GirlFragment;
import com.bzh.dytt.ui.AllMoviePageFragment;
import com.bzh.dytt.ui.ImdbMovieFragment;
import com.bzh.dytt.ui.LoadableMoviePageFragment;
import com.bzh.dytt.ui.NewMovieFragment;
import com.bzh.dytt.ui.VideoDetailPageFragment;
import com.bzh.dytt.search.SearchFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class FragmentBuildersModule {

    @ContributesAndroidInjector
    abstract AllMoviePageFragment contributeHomePageFragment();

    @ContributesAndroidInjector
    abstract ColorHuntFragment contributeColorHuntFragment();

    @ContributesAndroidInjector
    abstract GirlFragment contributeGirlFragment();

    @ContributesAndroidInjector
    abstract VideoDetailPageFragment contributeVideoDetailFragment();

    @ContributesAndroidInjector
    abstract NewMovieFragment contributeNewMovieFragment();

    @ContributesAndroidInjector
    abstract LoadableMoviePageFragment contributeLoadbleMoviePageFragment();

    @ContributesAndroidInjector
    abstract SearchFragment contributeSearchFragment();

    @ContributesAndroidInjector
    abstract ImdbMovieFragment contributeImdbMovieFragment();
}

File: app/src/main/java/com/bzh/dytt/di/FragmentBuildersModule.kt
package com.bzh.dytt.di

import com.bzh.dytt.ui.detail.DetailFragment
import com.bzh.dytt.ui.home.HomeFragment
import com.bzh.dytt.ui.home.HomeListFragment
import com.bzh.dytt.ui.search.SearchFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class FragmentBuildersModule {

    @ContributesAndroidInjector
    abstract fun contributeMainHomePageFragment(): HomeFragment

    @ContributesAndroidInjector
    abstract fun contributeHomeListPageFragment(): HomeListFragment

    @ContributesAndroidInjector
    abstract fun contributeDetailPageFragment(): DetailFragment

    @ContributesAndroidInjector
    internal abstract fun contributeSearchFragment(): SearchFragment
}
